File: Betreibssysteme I/Praktikum/062-fibonacci.java
class Fibonacci {
  int n;
  int i = 0;

  Fibonacci(int n) {
    this.n = n;

    for (int c = 0; c <= n; c++) {
      System.out.println(f(i));
      i++;
    }
  }
  
  public static void main(String args[]) {
    System.out.println("Fibonacci Folge ist:");

    Fibonacci fib = new Fibonacci(50);
  }

  private int f(int n) {
    if (n == 0 || n == 1) {
      return n;
    } else {
      return (f(n-1)+f(n-2));
    }
  }
}

File: Betreibssysteme I/Praktikum/062-fibonacci.c
#include <stdio.h>
#include <stdlib.h>

int fibonacci(int);

int main(int argc, char **argv)
{
  int n = atoi(argv[1]), c, i = 0;

  // scanf("%d", &n);

  printf("Fibonacci-Folge ist:\n");

  for (c = 1; c <= n; c++)
  {
    printf("%d\n", fibonacci(i));
    i++;
  }

  return 0;
}

int fibonacci(int n)
{
  if (n == 0 || n == 1) return n;
  else return (fibonacci(n-1)+fibonacci(n-2));
}
